### ISSUE TITLE
Initialized is fired too late?

### ISSUE THREAD

**Issue by tomv564** [tomv564/LSP#75]
LSP logs `initialized` after the first `textDocument/didOpen` goes out.
The spec says it should be sent: 
>  after the client received the result of the initialize request but before the client is sending any other request or notification to the server

It is possible the logging is out-of-order, need to investigate.

```
LSP: starting ['pyls']
LSP: request 1: initialize 
LSP: client registered for window 2 {'pyls': <LSP.main.Client object at 0x106a7a2d0>}
LSP: got json:  {"result":{"capabilities":{"definitionProvider":true,"signatureHelpProvider":{"triggerCharacters":["(",","]},"hoverProvider":true,"documentSymbolProvider":true,"executeCommandProvider":{"commands":[]}
LSP: notify: textDocument/didOpen
LSP: notify: initialized
LSP: got json:  {"method"
```

**Comment by rwols** [comment]
Yes, I also noticed didChange notifications to be somewhat out of date with me typing on my keyboard, but I assume this is by design?

**Comment by tomv564** [comment]
didChange is fired either 500ms after you've stopped typing, or immediately when a completion is requested. I hope the logging matches that behaviour :)

**Comment by rwols** [comment]
On another note, are you going to put more capabilities in the Initialize request?

**Comment by tomv564** [comment]
We must be missing a few since I last looked at it... it looks like `applyEdit` on workspace, `synchronization` on textDocument are incomplete.

**Comment by MichaelAllenHardeman** [comment]
@tomv564 I did not install via git, but I'd be happy to try it out. I don't see any documents that describe how to build the extension properly; can you point me in the right direction?

**Comment by MichaelAllenHardeman** [comment]
@tomv564 I ended up just dropping /plugins/core/windows.py into the .sublime-package in the right spot, and the `initialized` fired before the `didOpen` event. Everything worked as expected.
